lib/streams/compound-duplex.js

```diff
@@ -63,12 +63,40 @@ function CompoundDuplex(writable, readable, options) {
 		self.push(null);
 	});
 
-	this.on('finish', function() {
-		writable.end();
+	readable.on('finish', function() {
```

Comment:
Good call. I actually didn't need any of the event listeners at the end of the constructor.